great_expectations/data_context/data_context.py

```diff
@@ -378,57 +381,71 @@ def get_docs_sites_urls(self, resource_identifier=None):
-                    site_urls.append({
-                        "site_name": site_name,
-                        "site_url": url
-                    })
+        if site_name:
+            site = sites[site_name]
+            site_builder = self._load_site_builder_from_site_config(site)
+            url = site_builder.get_resource_url(resource_identifier=resource_identifier)
+            return [{"site_name": site_name, "site_url": url}]
+
+        site_urls = []
+        for _site_name, site_config in sites.items():
+            site_builder = self._load_site_builder_from_site_config(site_config)
+            url = site_builder.get_resource_url(resource_identifier=resource_identifier)
+            site_urls.append({"site_name": _site_name, "site_url": url})
 
         return site_urls
 
+    def _load_site_builder_from_site_config(self, site_config) -> SiteBuilder:
+        default_module_name = "great_expectations.render.renderer.site_builder"
+        site_builder = instantiate_class_from_config(
+            config=site_config,
+            runtime_environment={
+                "data_context": self,
+                "root_directory": self.root_directory,
+            },
+            config_defaults={"module_name": default_module_name},
+        )
+        if not site_builder:
+            raise ge_exceptions.ClassInstantiationError(
+                module_name=default_module_name,
+                package_name=None,
+                class_name=site_config["class_name"],
+            )
+        return site_builder
+
     @usage_statistics_enabled_method(event_name="data_context.open_data_docs",)
-    def open_data_docs(self, resource_identifier=None):
+    def open_data_docs(self, resource_identifier=None, site_name=None):
         """
         A stdlib cross-platform way to open a file in a browser.
 
         :param resource_identifier: ExpectationSuiteIdentifier, ValidationResultIdentifier
                 or any other type's identifier. The argument is optional - when
                 not supplied, the method returns the URL of the index page.
+        :param site_name: Optionally specify which site to open. If not
+                specified, open all docs found in the project.
         """
-        data_docs_urls = self.get_docs_sites_urls(resource_identifier=resource_identifier)
-        for site_dict in data_docs_urls:
-            logger.debug("Opening Data Docs found here: {}".format(site_dict["site_url"]))
-            webbrowser.open(site_dict["site_url"])
+        data_docs_urls = self.get_docs_sites_urls(
+            resource_identifier=resource_identifier,
+            site_name=site_name,
+        )
+        urls_to_open = [site["site_url"] for site in data_docs_urls]
+
+        if site_name:
+            for site in data_docs_urls:
+                if site["site_name"] == site_name:
+                    urls_to_open = [site["site_url"]]
```

Comment:
Didn't `get_docs_sites_urls` just filter by `site_name`?

Reply:
Lol yup. Can you tell which one I did first! Cleaning this up.